frontend/src/metabase/query_builder/components/CompareAggregations/CompareAggregations.tsx

```diff
@@ -41,13 +44,27 @@ export const CompareAggregations = ({
   >(hasManyAggregations ? undefined : aggregations[0]);
   const [offset, setOffset] = useState<number | "">(DEFAULT_OFFSET);
   const [columns, setColumns] = useState<ColumnType[]>(DEFAULT_COLUMNS);
+  const [comparisonType, setComparisonType] = useState<ComparisonType>(
+    DEFAULT_COMPARISON_TYPE,
+  );
+  const [includeCurrentPeriod, setIncludeCurrentPeriod] = useState(false);
   const width = aggregation ? STEP_2_WIDTH : STEP_1_WIDTH;
 
   const title = useMemo(
     () => getTitle(query, stageIndex, aggregation),
     [query, stageIndex, aggregation],
   );
 
+  const handleComparisonTypeChange = useCallback(
+    (comparisonType: ComparisonType) => {
+      setComparisonType(comparisonType);
+      if (comparisonType === "moving-average" && offset === 1) {
```

Comment:
<img width="642" alt="Screenshot 2024-08-13 at 10 28 36" src="https://github.com/user-attachments/assets/241b842a-6041-459b-8368-e8f961d00ad8">

Try changing it to `1` afterwards. I think we should coerce it to 2 automatically and don't show this error.